### DISCUSSION TITLE
Can turborepo detects binaries in downstream folder, for upstream folder

### DISCUSSION THREAD

**Discussion by igortas** [vercel/turborepo#461] [Help]
Hi,

When I run turbo run dev from root package, I like to execute in parallel two dev:* scripts in parallel in apps/remix folder, but concurrently is not found.

Both packages are imported as:
```
"config": "workspace:*",
"shared": "workspace:*"
```

I have two different subfolders in packages folder like:
  - shared (package.json with most utils packages like concurrently, rimraf installed as dev deps)
  - config (eslint configuration, with package.json and eslint installed as dev deps)

Checked node_modules in /apps/remix and both binary exists inside ./node_modules/config or ./node_modules/shared symlinks

So my question is, can be something done more composable, like apps/A, can invoke packages/a binaries, like rimraf, concurrency installed in packages/a, but for example apps/B, don't need anything from packages/b.
Other candidate is react, react-dom, where if I have UI lib, i like everything to be there, cuz If I have multiple apps that need my custom lib, probably all apps, will use the same react version.

Now, to make it run everything is hoisted in root package.json 

Commands:
```
"dev": "concurrently \"pnmp dev:remix\" \"pnpm dev:miniflare\"",
"dev:remix": "remix watch",
"dev:miniflare": "miniflare --build-command \"pnpx ts-node ./build.ts\" --disable-cache --watch",
"lint": "eslint src --fix",
```

Error:
```
remix:dev: cache miss, executing e7b2b9bfaf1b0e0d
remix:dev: > concurrently "pnmp dev:remix" "pnpm dev:miniflare"

remix:dev:  ELIFECYCLE  Command failed.
remix:dev: sh: 1: concurrently: not found
remix:dev: command finished with error, but continuing...
```

BR, Igor

**Comment by anthonyshew** [comment]
Invoking binaries from places that they don't exist probably isn't a great thing for `turbo` to implement. There are more favorable ways that one can architect your repository using Turborepo's primitives.
